### ISSUE TITLE
Replace zerolog by slog

### ISSUE THREAD

**Issue by goloroden** [thenativeweb/eventsourcingdb-client-golang#79]
### What is this task about?

The upcoming version 1.21 of Go includes a new package [log/slog](https://pkg.go.dev/log/slog) for structured logging. As we have already done in the ESDB server, we can (and should) replace zerolog by this module.

### What needs to be done to complete this task?

- [ ] Replace zerolog by slog

### What else should we know?

Since the `slog` package is only included in Go 1.21, we should do this only after Go 1.21 was released.
